rubrik_cloudout_consolidation.template

```diff
@@ -0,0 +1,511 @@
+                "s3:ListAllMyBuckets"
+              ],
+              "Resource": "*"
+            }
+          ]
+        },
+        "Users": [{ "Ref": "IAMUserName" }]
+      }
+    },
+    
+    "CreateKMSKey": {
+      "Type": "AWS::KMS::Key",
+      "Condition": "CreateKMS",
+      "Properties": {
+        "Description": "Rubrik archive location encryption",
+        "KeyPolicy": {
+          "Version": "2012-10-17",
+          "Id": {
+            "Fn::Join": ["", ["rubrik-encryption-", { "Ref": "S3BucketName" }]]
+          },
+          "Statement": [
+            {
+              "Sid": "Enable IAM User Permissions",
+              "Effect": "Allow",
+              "Principal": {
+                "AWS": {
+                  "Fn::Join": [
+                    "",
+                    ["arn:aws:iam::", { "Ref": "AWS::AccountId" }, ":root"]
+                  ]
+                }
+              },
+              "Action": "kms:*",
+              "Resource": "*"
+            },
+            {
+              "Sid": "Allow use of the key",
+              "Effect": "Allow",
+              "Principal": {
+                "AWS": {
+                  "Fn::If": [
+                    "CreateIAMuser",
+                    {
+                      "Fn::GetAtt": ["CreateNewIAMUser", "Arn"]
+                    },
+                    {
+                      "Fn::Join": [
+                        "",
+                        [
+                          "arn:aws:iam::",
+                          { "Ref": "AWS::AccountId" },
+                          ":user/",
+                          { "Ref": "IAMUserName" }
+                        ]
+                      ]
+                    }
+                  ]
+                }
+              },
+              "Action": [
+                "kms:Encrypt",
+                "kms:Decrypt",
+                "kms:ReEncrypt*",
+                "kms:GenerateDataKey*",
+                "kms:DescribeKey"
+              ],
+              "Resource": "*"
+            }
+          ]
+        }
+      }
+    },
+
+    "KMSKeyAlias": {
+      "Type": "AWS::KMS::Alias",
+      "DependsOn": "CreateKMSKey",
+      "Condition": "CreateKMS",
+      "Properties": {
+        "AliasName": {
+          "Fn::Join": [
+            "",
+            ["alias/rubrik-encryption-", { "Ref": "S3BucketName" }]
+          ]
+        },
+        "TargetKeyId": { "Ref": "CreateKMSKey" }
+      }
+    }
+  },
+
+  "Outputs": {
+    "AWSBucketName": {
+      "Description": "The S3 Bucket name.",
+      "Value": { "Ref": "S3BucketName" }
+    },
+    "IAMUserAccessKey": {
+      "Condition": "CreateIAMuser",
+      "Description": "Access Key for the new IAM User.",
+      "Value": { "Ref": "IAMUserAccessKeys" }
+    },
+    "IAMUserSecretKey": {
+      "Condition": "CreateIAMuser",
+      "Description": "Secret Key for the new IAM user.",
+      "Value": { "Fn::GetAtt": ["IAMUserAccessKeys", "SecretAccessKey"] }
+    },
+    "KMSKeyId": {
```

Comment:
Modern best practices dictate that it would be better to store this key in the AWS secrets manager. Leaving them in the CFT makes them show up in a public manner. See https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-secretsmanager-secret.html for details on how to create secrets in the AWS Secrets Manger. You can use the KMS Key that's generated by this CFT to encrypt the secret.

Reply:
This output should be replaced with one to check the Secrets manager.

Reply:
Added #18 to resolve this in teh future.